mypy/stubutil.py

```diff
@@ -240,11 +241,8 @@ def remove_misplaced_type_comments(source: Union[str, bytes]) -> Union[str, byte
 def common_dir_prefix(paths: List[str]) -> str:
     if not paths:
         return '.'
-    cur = os.path.dirname(paths[0])
-    for path in paths[1:]:
-        while True:
-            path = os.path.dirname(path)
-            if (cur + '/').startswith(path + '/'):
-                cur = path
-                break
-    return cur or '.'
+    cur = Path(paths[0])
+    for parent in cur.parents:
+        if all(parent in Path(path).parents for path in paths[1:]):
+            return str(parent)
+    return '.'
```

Comment:
I'd rather not introduce code using `pathlib`, since there is a lot of code using `os.path` elsewhere. I know that `pathlib` has various improvements over `os.path`, but I think that consistency is more important. Could you implement the fix using `os.path`?

Reply:
Sure, can do. However, do you find the idea of switching to `pathlib` useful in general? If yes, I could replace all the `os.path` usages with appropriate `pathlib` calls throughout the codebase in a separate PR.

Reply:
Thanks for the offer, but I'm worried that it might introduce subtle bugs since some of the path handling code is pretty delicate. (It looks like we have a few references to `pathlib` already, though.)

Reply:
Was worth a shot :-)